central/reports/scheduler/v2/schedule.go

```diff
@@ -0,0 +1,624 @@
+                             discoveredAtImage
+		                     link
+                         }`
+
+	deployedImagesReportQuery = `query getDeployedImagesReportData($scopequery: String, 
+                               $cvequery: String, $pagination: Pagination) {
+							       deployments: deployments(query: $scopequery, pagination: $pagination) {
+                                       clusterName
+								       namespace
+                                       name
+                                       images {
+                                           name {
+								               full_name:fullName
+								           }
+                                           imageComponents {
+									           name
+									           imageVulnerabilities(query: $cvequery) {
+										           ...cveFields
+									           }
+								           }
+							           }
+						           }
+					           }` +
+		cveFieldsFragment
+	deployedImagesReportQueryOpName = "getDeployedImagesReportData"
+
+	watchedImagesReportQuery = `query getWatchedImagesReportData($scopequery: String, $cvequery: String, $pagination: Pagination) {
+                              images: images(query: $scopequery, pagination: $pagination) {
+                                  name {
+                                      full_name:fullName
+                                  }
+                                  imageComponents {
+                                      name
+                                      imageVulnerabilities(query: $cvequery) {
+                                          ...cveFields
+                                      }
+                                  }
+                              }
+                          }` +
+		cveFieldsFragment
+	watchedImagesReportQueryOpName = "getWatchedImagesReportData"
+
+	vulnReportEmailTemplate = `
+	{{.BrandedProductName}} has found vulnerabilities associated with the running container images owned by your organization. Please review the attached vulnerability report {{.WhichVulns}} for {{.DateStr}}.
+
+	To address these findings, please review the impacted software packages in the container images running within deployments you are responsible for and update them to a version containing the fix, if one is available.`
+
+	noVulnsFoundEmailTemplate = `
+	{{.BrandedProductName}} has found zero vulnerabilities associated with the running container images owned by your organization.`
+
+	paginatedQueryStartOffset = 0
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	scheduledCtx = resolvers.SetAuthorizerOverride(loaders.WithLoaderContext(sac.WithAllAccess(context.Background())), allow.Anonymous())
+)
+
+// Scheduler maintains the schedules for reports
+type Scheduler interface {
+	UpsertReportSchedule(reportConfig *storage.ReportConfiguration) error
+	RemoveReportSchedule(reportConfigID string)
+	SubmitReport(request *ReportRequest, reSubmission bool) (string, error)
```

Comment:
`SubmitReportJob` or `SubmitReportRequest` since it is not a report that is being submitted but a report request